framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,293 @@
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.__workingDir = runInfo['WorkingDir']
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs
+      @ In, xmlNode, xml.etree.ElementTree Element Objects, the xml element node that will be checked against the available options specific to this Sampler
+      @ Out, None
+    """
+    paramInput = Metric.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+
+    for child in paramInput.subparts:
+      if child.getName() == 'Metric':
+        if 'type' not in child.parameterValues.keys() or 'class' not in child.parameterValues.keys():
+          self.raiseAnError(IOError, 'Tag Metric must have attributes "class" and "type"')
+        else:
+          metricName = child.value.strip()
+          self.metricsDict[metricName] = None
+      elif child.getName() == 'Features':
+        self.features = list(var.strip() for var in child.value.split(','))
+        self.featuresType = child.parameterValues['type']
+      elif child.getName() == 'Targets':
+        self.targets = list(var.strip() for var in child.value.split(','))
+        self.TargetsType = child.parameterValues['type']
+      else:
+        self.raiseAnError(IOError, "Unknown xml node ", child.getName(), " is provided for metric system")
+
+    if not self.features:
+      self.raiseAnError(IOError, "XML node 'Features' is required but not provided")
+    elif len(self.features) != len(self.targets):
+      self.raiseAnError(IOError, 'The number of variables found in XML node "Features" is not equal the number of variables found in XML node "Targets"')
+
+  def collectOutput(self,finishedJob, output):
+    """
+      Function to place all of the computed data into the output object, (Files or DataObjects)
+      @ In, finishedJob, object, JobHandler object that is in charge of running this postprocessor
+      @ In, output, object, the object where we want to place our computed results
+      @ Out, None
+    """
+    if finishedJob.getEvaluation() == -1:
+      self.raiseAnError(RuntimeError, ' No available output to collect')
+    outputDict = finishedJob.getEvaluation()[1]
+
+    if isinstance(output, Files.File):
+      availExtens = ['xml', 'csv']
+      outputExtension = output.getExt().lower()
+      if outputExtension not in availExtens:
+        self.raiseAMessage('Metric postprocessor did not recognize extension ".', str(outputExtension), '". The output will be dumped to a text file')
+      output.setPath(self.__workingDir)
+      self.raiseADebug('Write Metric prostprocessor output in file with name: ', output.getAbsFile())
+      output.open('w')
+      if outputExtension == 'xml':
+        self._writeXML(output, outputDict)
+      else:
+        separator = ' ' if outputExtension != 'csv' else ','
+        self._writeText(output, outputDict, separator)
```

Comment:
probably you can just error out here and remove (or comment out) the ```_writeText``` method below

Reply:
Commented out.